distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/JvmOptionsParserTests.java

```diff
@@ -344,4 +349,27 @@ public void accept(final int lineNumber, final String line) {
         assertThat(seenInvalidLines, equalTo(invalidLines));
     }
 
+    public void testNodeProcessorsActiveCount() {
+        {
+            final List<String> jvmOptions = SystemJvmOptions.systemJvmOptions(Settings.EMPTY);
+            assertThat(jvmOptions, not(hasItem(containsString("-XX:ActiveProcessorCount="))));
+        }
+        {
+            Settings nodeSettings = Settings.builder().put(EsExecutors.NODE_PROCESSORS_SETTING.getKey(), 1).build();
+            final List<String> jvmOptions = SystemJvmOptions.systemJvmOptions(nodeSettings);
+            assertThat(jvmOptions, hasItem("-XX:ActiveProcessorCount=1"));
+        }
+        {
+            // check rounding
+            Settings nodeSettings = Settings.builder().put(EsExecutors.NODE_PROCESSORS_SETTING.getKey(), 0.5).build();
```

Comment:
nit: let us check the round up using 0.2 instead?

```suggestion
            Settings nodeSettings = Settings.builder().put(EsExecutors.NODE_PROCESSORS_SETTING.getKey(), 0.2).build();
```